spec/unit/transaction_spec.rb

```diff
@@ -896,6 +897,30 @@ def post_resource_eval
       expect(transaction.report.resource_statuses['Notify[b]']).to be_skipped
     end
   end
+
+  describe "failed dependency is depended on multiple times" do
+    it "notifies the failed dependency once" do
+      Puppet::Util::Execution.stubs(:execute).with(["/my/command"]).raises(Puppet::ExecutionFailure, "Failed")
+
+      Puppet::Type::Notify.any_instance.expects(:send_log).with(:notice, "Dependency Exec[exec1] has failures: true")
+      Puppet::Type::Notify.any_instance.expects(:send_log).with(:notice, "Dependency Exec[exec2] has failures: true")
+      Puppet::Type::Notify.any_instance.expects(:send_log).with(:notice, "Dependency Exec[exec3] has failures: true")
+      Puppet::Type::Notify.any_instance.expects(:send_log).with(:notice, "Dependency Exec[exec4] has failures: true")
+      Puppet::Type::Notify.any_instance.expects(:send_log).with(:notice, "Dependency Exec[exec5] has failures: true")
+
+      Puppet::Type::Notify.any_instance.expects(:send_log).with(:warning, "Skipping because of failed dependencies").times(3)
+
+      command_string = File.expand_path('/my/command')
```

Comment:
I'm surprised this passes on Windows, because here `command_string` will be something like `C:/my/command`, but the `Execution.stubs(:execute)` above only matches `/my/command`. Probably better to move this `command_string` before the `stubs` line, and use the former as an argument of what to stub.